mm2src/mm2_main/src/lp_ordermatch.rs

```diff
@@ -2049,31 +2075,66 @@ impl MakerOrder {
             return OrderMatchResult::NotMatched;
         }
 
+        let is_legacy = self.swap_version.is_legacy() || taker.swap_version.is_legacy();
+
```

Comment:
nit: we may get premium as reference here, like:
```
        let premium_default = MmNumber::default();
        let premium = self.premium.as_ref().unwrap_or(&premium_default);
```
and use it within `match taker.action {...}` instead of cloning.
(but not sure this is much better though)